### DISCUSSION TITLE
open BOOTSEL via software?

### DISCUSSION THREAD

**Discussion by LazaroFilm** [earlephilhower/arduino-pico#1796] [General]
Hi, is there a way to load the BOOTSEL to a computer via code only? I have a pi pico with touchscreen and want to make a menu button that says Firmware update. When clicked it opens the folder on the desktop to load the new firmware in. This would streamline sending updates to clients. I don't have any physical buttons available on the device, only touchscreen

SOLVED: 
calling this function anywhere in your code resets the board into Bootsel: `reset_usb_boot(0, 0);`
Then drag/drop a new *.uf2 firmware into the folder or disconnecting/reconnecting the USB restarts to the regular mode.

**Comment by earlephilhower** [comment]
Check the Rp2040 docs in this repo.  There's a reset to bootloader method which should do this.

**Reply by LazaroFilm** [reply]
Thanks! This put me on the right path to a solution!
